Code/GraphMol/ChemReactions/SanitizeRxn.cpp

```diff
@@ -324,15 +323,67 @@ void fixReactantTemplateAromaticity(ChemicalReaction &rxn) {
 }
 
 void fixHs(ChemicalReaction &rxn) {
+  {
+    //  if mapped Hydrogens in reactants area mapped to heavy atoms
+    //   keep mappings, in all other cases remove them.
+    //   this allows us to merge query hydrogens atoms 
+
+    std::map<int, bool> mappedToNonHeavyProductAtom;
```

Comment:
I just went to implement this and released we need to know the max atom map number or default the bitset to dynamic_bitset<> mappedNonHeavyProductAtoms(999).  I'm not sure doing two passes on the atoms is worth this optimization.